### PR TITLE
Creating DI Extension + .NET Framework assembly directory format fix

### PR DESCRIPTION
**Pull request by Basyras**
**Creating Extension that allows inject all RazorLightEngine dependecies**
Found bug when using .Net Framework project  (#355)  wich can be easily solved when dependency injection is used to create RazorLightEngine 
so ... i did it?
**Usage**

			services.AddRazorLight()
				.UseMemoryCachingProvider()
				.UseFileSystemProject(root)
				.UseNetFrameworkLegacyFix();

### CODE REVIEW

File: src/RazorLight/Compilation/DefaultAssemblyDirectoryFormatter.cs [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+	public class DefaultAssemblyDirectoryFormatter : IAssemblyDirectoryFormatter
+	{
+		public string GetAssemblyDirectory(Assembly assembly)
+		{
+			string codeBase = assembly.CodeBase;
+			UriBuilder uri = new UriBuilder(codeBase);
+			return Uri.UnescapeDataString(uri.Path);
+		}
+	}
+}
```
Comment (Basyras) [inline src/RazorLight/Compilation/DefaultAssemblyDirectoryFormatter.cs:17]: extracting old DefaultMetadataReferenceManagaer AssemblyDirectory(Assembly assembly) method into injectable service

File: src/RazorLight/RazorLightOptions.cs [hunk 2]
```diff
@@ -30,6 +31,8 @@ public RazorLightOptions()
 
 		public ICachingProvider CachingProvider { get; set; }
 
+		public Assembly OperatingAssemlby { get; set; }
```
Comment (jzabroski) [inline src/RazorLight/RazorLightOptions.cs]: Typo.  Should say OperatingAssembly.

File: src/RazorLight/Compilation/RoslynCompilationService.cs [hunk 3]
```diff
@@ -41,6 +41,11 @@ public RoslynCompilationService(IMetadataReferenceManager referenceManager, Asse
 			EmitOptions = new EmitOptions(debugInformationFormat: pdbFormat);
 		}
 
+		public RoslynCompilationService(IMetadataReferenceManager referenceManager, IOptions<RazorLightOptions> options) :this(referenceManager,options.Value.OperatingAssemlby)
```
Comment (jzabroski) [inline src/RazorLight/Compilation/RoslynCompilationService.cs]: typo.  Should say OperatingAssembly

File: src/RazorLight/Extensions/ServiceCollectionExtensions.cs [hunk 4]
```diff
@@ -30,6 +36,79 @@ public static void AddRazorLight(this IServiceCollection services, Func<IRazorLi
...
+
+
+
+		///// <summary>
+		///// Adds required RazorLightProject
+		///// </summary>
+		///// <param name="services"></param>
+		///// <param name="root">Directory path to the root folder containing your Razor markup files.</param>
+		///// <param name="extension">If you wish, you can use a different extension than .cshtml.</param>
+		//public static void AddRazorLightFileSystemProject(this IServiceCollection services,string root, string extension = null)
```
Comment (jzabroski) [inline src/RazorLight/Extensions/ServiceCollectionExtensions.cs:74]: Why is all this code commented out?

Comment (Basyras) [reply src/RazorLight/Extensions/ServiceCollectionExtensions.cs:74]: @jzabroski To be honest i just forgot to delete it. It was a initial way how to inject services but i rewrote everthing so the exact logic is now used inside RazorLightDependecyBuilder wich i find more user friendly. So if the RazorLightDependecyBuilder class is ok those comments should be deleted.

**Pull request merged**